crates/uv/tests/pip_tree.rs

```diff
@@ -1500,3 +1499,249 @@ fn package_flag() {
+    │   └── trove-classifiers v2024.3.3 [required: *]
+    ├── msgspec v0.18.6 [required: >=0.18.4, <0.19.0]
+    ├── setuptools v69.2.0 [required: >=69.1.1, <70.0.0]
+    └── twine v4.0.2 [required: >=4.0.2, <5.0.0]
+        ├── pkginfo v1.10.0 [required: >=1.8.1]
+        ├── readme-renderer v43.0 [required: >=35.0]
+        │   ├── nh3 v0.2.15 [required: >=0.2.14]
+        │   ├── docutils v0.20.1 [required: >=0.13.1]
+        │   └── pygments v2.17.2 [required: >=2.5.1]
+        ├── requests v2.31.0 [required: >=2.20]
+        │   ├── charset-normalizer v3.3.2 [required: <4, >=2]
+        │   ├── idna v3.6 [required: <4, >=2.5]
+        │   ├── urllib3 v2.2.1 [required: <3, >=1.21.1]
+        │   └── certifi v2024.2.2 [required: >=2017.4.17]
+        ├── requests-toolbelt v1.0.0 [required: !=0.9.0, >=0.8.0]
+        │   └── requests v2.31.0 [required: <3.0.0, >=2.0.1] (*)
+        ├── urllib3 v2.2.1 [required: >=1.26.0]
+        ├── importlib-metadata v7.1.0 [required: >=3.6]
+        │   └── zipp v3.18.1 [required: >=0.5]
+        ├── keyring v25.0.0 [required: >=15.1]
+        │   ├── jaraco-classes v3.3.1 [required: *]
+        │   │   └── more-itertools v10.2.0 [required: *]
+        │   ├── jaraco-functools v4.0.0 [required: *]
+        │   │   └── more-itertools v10.2.0 [required: *]
+        │   └── jaraco-context v4.3.0 [required: *]
+        ├── rfc3986 v2.0.0 [required: >=1.4.0]
+        └── rich v13.7.1 [required: >=12.0.0]
+            ├── markdown-it-py v3.0.0 [required: >=2.2.0]
+            │   └── mdurl v0.1.2 [required: ~=0.1]
+            └── pygments v2.17.2 [required: >=2.13.0, <3.0.0]
+    (*) Package tree already displayed
+
+    ----- stderr -----
+    "###
+    );
+}
+
+#[test]
+fn emit_version_specifier_with_invert() {
+    let context = TestContext::new("3.12");
+
+    let requirements_txt = context.temp_dir.child("requirements.txt");
+    requirements_txt
+        .write_str("scikit-learn==1.4.1.post1")
+        .unwrap();
+
+    uv_snapshot!(context
+        .pip_install()
+        .arg("-r")
+        .arg("requirements.txt")
+        .arg("--strict"), @r###"
+    success: true
+    exit_code: 0
+    ----- stdout -----
+
+    ----- stderr -----
+    Resolved 5 packages in [TIME]
+    Prepared 5 packages in [TIME]
+    Installed 5 packages in [TIME]
+     + joblib==1.3.2
+     + numpy==1.26.4
+     + scikit-learn==1.4.1.post1
+     + scipy==1.12.0
+     + threadpoolctl==3.4.0
+    "###
+    );
+
+    uv_snapshot!(
+        context.filters(),
+        context.pip_tree()
+        .arg("--emit-version-specifier")
+        .arg("--invert"), @r###"
+    success: true
+    exit_code: 0
+    ----- stdout -----
+    joblib v1.3.2
+    └── scikit-learn v1.4.1.post1 [requires: joblib>=1.2.0]
+    numpy v1.26.4
+    ├── scikit-learn v1.4.1.post1 [requires: numpy<2.0, >=1.19.5]
+    └── scipy v1.12.0 [requires: numpy<1.29.0, >=1.22.4]
+        └── scikit-learn v1.4.1.post1 [requires: scipy>=1.6.0]
```

Comment:
```suggestion
        └── scikit-learn v1.4.1.post1 [requires: scipy >=1.6.0]
```
feeling similarly conflicted about the lack of space after the package name here

Reply:
Purely from feeling i think space should be emited :)